### PR TITLE
Fix Bisq 2 onion-grater profile

### PR DESCRIPTION
**Pull request by alvasw**
- Remove deprecated SETEVENTS command
- Add missing ADD_ONION pattern and replacement
- Fix HS_DESC UPLOAD pattern and replacement
- Add missing HS_DESC FAILED pattern and replacement

Ref: bisq-network/bisq2#1894

### CODE REVIEW

**Comment by alvasw** [comment]
Hi @adrelanos,

I'm one of the Bisq core contributors and worked on the Bisq 2 Whonix support over the last few weeks. Bisq used the jtorctl library from the Tor Project [1] until now. Unforunately, is is not maintained by the Tor Project anymore and outdated. We re-implemented the Tor Control Protocol in Java to add Whonix/onion-grater support.

This commit and the commits in Bisq are signed with the same GPG key (see [2] and [3]).

[1] https://github.com/torproject/jtorctl
[2] https://github.com/bisq-network/bisq/pulls?q=is%3Apr+author%3Aalvasw+
[3] https://github.com/bisq-network/bisq2/pulls?q=is%3Apr+author%3Aalvasw+

File: usr/share/doc/onion-grater-merger/examples/40_bisq.yml [hunk 1]
```diff
@@ -60,8 +60,10 @@
       response:
         - pattern:     '650 HS_DESC CREATED (\S+) (\S+) (\S+) \S+ (.+)'
           replacement: '650 HS_DESC CREATED {} {} {} redacted {}'
-        - pattern:     '650 HS_DESC UPLOAD (\S+) (\S+) .*'
-          replacement: '650 HS_DESC UPLOAD {} {} redacted redacted'
+        - pattern:     '650 HS_DESC FAILED (\S+) \S+ \S+ \S+ (.+)'
+          replacement: '650 HS_DESC FAILED {} redacted redacted redacted {}'
+        - pattern:     '650 HS_DESC UPLOAD (\S+) (\S+) (\S+) (\S+) (\S+)'
+          replacement: '650 HS_DESC UPLOAD {} {} {} {} {}'
```
Comment (nyxnor) [inline usr/share/doc/onion-grater-merger/examples/40_bisq.yml:66]: Are all those fields in `HS_DESC UPLOAD` required to be seen by the Bisq client?

[Cwtch can use less information](https://github.com/Whonix/onion-grater/blob/master/usr/share/doc/onion-grater-merger/examples/40_cwtch.yml#L71-L72) and [Onionshare also](https://github.com/Whonix/onion-grater/blob/master/usr/share/doc/onion-grater-merger/examples/40_onionshare.yml#L89-L90).

**Review by nyxnor — CHANGES_REQUESTED** [review]

**Comment by nyxnor** [comment]
Thanks for contributing!

**Comment by nyxnor** [comment]
For this PR to get merged, we need to test Bisq2 release greater than 2.0.4. https://github.com/bisq-network/bisq2/issues/1894#issuecomment-2164679353

**Comment by alvasw** [comment]
Please apologize my late reply!

> Are all those fields in HS_DESC UPLOAD required to be seen by the Bisq client?

Yes, it's used as a DoS defense in the handshake. When a Bisq client receives an incoming connection from another peer, the connecting peer proves its onion address ownership and the other peer verifies that the onion address is online by checking whether the onion service's hidden service descriptors were uploaded.

So if Alice connects to Bob before Alice has uploaded her hidden service descriptors, Bob will reject the connection.

We could count the number of hidden service descriptor uploads and successes/failures instead of verifying the directories.

File: usr/share/doc/onion-grater-merger/examples/40_bisq.yml (hunk 1 above)
Comment (alvasw) [reply usr/share/doc/onion-grater-merger/examples/40_bisq.yml:66]: Yes, it's used as a DoS defense in the handshake. When a Bisq client receives an incoming connection from another peer, the connecting peer proves its onion address ownership and the other peer verifies that the onion address is online by checking whether the onion service's hidden service descriptors were uploaded.

So if Alice connects to Bob before Alice has uploaded her hidden service descriptors, Bob will reject the connection.

**Comment by nyxnor** [comment]
Tested on Bisq2.1.0. It is working!

**Comment by adrelanos** [comment]
Thank you!

**Pull request merged**